### PR TITLE
Implement react hooks

### PR DESCRIPTION
**Pull request by aguscha333**
## Implement react hooks

This PR implements hooks in the native base, this is a port of the work done by @felipegarcia92 in the web base.

### Tested on
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/containers/MainScreen/index.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-  <View style={styles.container}>
-    <Text>Hey {email}, you&#39;re logged in!</Text>
-    <Button
-      onPress={logout}
-      title={translate('MAIN_SCREEN.logout')}
-    />
-  </View>
-);
+const MainScreen = () => {
+  const dispatch = useDispatch();
```
Comment (agustito37) [inline src/containers/MainScreen/index.js]: A small detail. Can you add a line as you do in the signup screen?

**Comment by peter7z** [comment]
Great job, thanks! It's awesome to have this in native too.
As a side note:
The app class is not a React component and it's in the file where the store is defined. I am not sure we should add hooks there, they might not even work outside of a component.

**Comment by mcousillas6** [comment]
> Great job, thanks! It's awesome to have this in native too.
> As a side note:
> The app class is not a React component and it's in the file where the store is defined. I am not sure we should add hooks there, they might not even work outside of a component.

Yes, if the App class is not a react component already, we don't need to migrate it, since hooks only work for functional components.

File: src/containers/MainScreen/index.js (hunk 1 above)
Comment (aguscha333) [reply src/containers/MainScreen/index.js]: fixed!

**Pull request merged**